packages/@aws-cdk/toolkit-lib/lib/toolkit/toolkit.ts

```diff
@@ -375,6 +378,85 @@ export class Toolkit extends CloudAssemblySourceBuilder {
+    const selectStacks = options.stacks ?? ALL_STACKS;
+    await using assembly = await assemblyFromSource(ioHelper, cx);
+    const stacks = await assembly.selectStacksV2(selectStacks);
+
+    let drifts = 0;
+    let uncheckedResources = 0;
+    const output = [];
+
+    for (const stack of stacks.stackArtifacts) {
+      const cfn = (await (await this.sdkProvider('drift')).forEnvironment(stack.environment, Mode.ForReading)).sdk.cloudFormation();
+
+      const driftResults = await detectStackDrift(
+        cfn,
+        ioHelper,
+        stack.stackName,
+      );
+
+      // Get all stack resources if we want to show all
+      const allStackResources = new Map<string, string>();
+      Object.keys(stack.template.Resources || {}).forEach(id => {
+        const resource = stack.template.Resources[id];
+        allStackResources.set(id, resource.Type);
+      });
+
+      if (!driftResults) {
+        await ioHelper.notify(IO.CDK_TOOLKIT_I4505.msg(`${chalk.bold(stack.displayName)}: unable to detect drift`));
+        continue;
+      }
+
+      const formatter = new DriftFormatter({
+        ioHelper: asIoHelper(this.ioHost, 'drift'),
+        stack,
+        driftResults,
+        allStackResources,
+      });
+
+      const driftOutput = formatter.formatStackDrift();
+
+      if (driftOutput.formattedDrift.stackHeader) {
+        await ioHelper.notify(IO.CDK_TOOLKIT_I4500.msg(driftOutput.formattedDrift.stackHeader, driftOutput));
+        output.push(driftOutput.formattedDrift.stackHeader);
+      }
+      if (driftOutput.formattedDrift.unchanged) {
+        await ioHelper.notify(IO.CDK_TOOLKIT_I4501.msg(driftOutput.formattedDrift.unchanged, driftOutput));
+        output.push(driftOutput.formattedDrift.unchanged);
+      }
+      if (driftOutput.formattedDrift.unchecked) {
+        await ioHelper.notify(IO.CDK_TOOLKIT_I4502.msg(driftOutput.formattedDrift.unchecked, driftOutput));
+        output.push(driftOutput.formattedDrift.unchecked);
+      }
+      if (driftOutput.formattedDrift.modified) {
+        await ioHelper.notify(IO.CDK_TOOLKIT_I4503.msg(driftOutput.formattedDrift.modified, driftOutput));
+        output.push(driftOutput.formattedDrift.modified);
+      }
+      if (driftOutput.formattedDrift.deleted) {
+        await ioHelper.notify(IO.CDK_TOOLKIT_I4504.msg(driftOutput.formattedDrift.deleted, driftOutput));
+        output.push(driftOutput.formattedDrift.deleted);
+      }
+      if (driftOutput.formattedDrift.finalResult) {
+        await ioHelper.notify(IO.CDK_TOOLKIT_I4500.msg(driftOutput.formattedDrift.finalResult, driftOutput));
+        output.push(driftOutput.formattedDrift.finalResult);
+      }
+      drifts += driftOutput.numResourcesWithDrift === -1 ? 0 : driftOutput.numResourcesWithDrift || 0;
+      uncheckedResources += driftOutput.numResourcesUnchecked === -1 ? 0 : driftOutput.numResourcesUnchecked || 0;
+      output.push(driftOutput.formattedDrift);
+    }
```

Comment:
We don't need to send the same payload multiple times and also it's the same stuff as in the message already.

Header and footer (finalResult?) can just be default messages without code and payload, they are not super interesting. Let's just turn them all into default messages.

Reply:
When you say default messages, are you referring to making `CDK_TOOLKIT_I4500` take no payload? The default messages seem to have been removed from `IO`

Reply:
Nevermind, discovered it as `ioHelper.defaults.<loggingLevel>`